handlers/dashboard.go

```diff
@@ -319,13 +421,13 @@ func DashboardDataBalanceCombined(w http.ResponseWriter, r *http.Request) {
 	q := r.URL.Query()
 	validatorLimit := getUserPremium(r).MaxValidators
 
-	queryValidators, err := parseValidatorsFromQueryString(q.Get("validators"), validatorLimit)
+	queryValidatorIndices, _, err := parseValidatorsFromQueryString(q.Get("validators"), validatorLimit)
```

Comment:
Right, I didn't consider manual queries of these methods. I created a new method `handleValidatorsQuery` that I call everywhere except for 3 remaining `parseValidatorsFromQueryString` calls:

- APIDashboardDataBalance
- DashboardDataBalance
- RewardNotificationSubscribe

The two balance ones are for API/APP and I didn't want to change the functionality but I can just use `handleValidatorsQuery` if you want me to.
The notification subscription is a POST request and should already have the correct validators.
We don't use the results and just check for a parsing error.

Reply:
By "should already have the correct validators" do you mean that the function already validates the indices or that the request should already be sent correct. If it's the latter, how so? If non-existent indices are given in a manual query, will that be caught?

Regarding API changes, I think keeping old behavior is for the best. If they caused errors for pubkeys in the request beforehand, they should continue doing that.

Reply:
My point was more that the dashboard should transfer the correct validators (only indices) to the rewards page when you click the button on the dashboard. The `RewardNotificationSubscribe` method is then for the handling of the rewards (or income) page which probably has a few things to change that are beyond the scope of this issue.

Currently if you go to https://beaconcha.in/user/rewards and try to subscribe to a non-existent validator like 999999999 it will still add it to the list, which is apparently something we want to keep for now.

I guess what we want for now is to just throw an error if a validator is a pubkey like before my changes.